test/shared-class.test.js

```diff
@@ -29,11 +29,6 @@ describe('SharedClass', function() {
       expect(sc.http.path).to.equal('/some');
     });
 
-    it('fills http.path using a normalized path', function() {
-      var sc = new SharedClass('SomeClass', SomeClass, {normalizeHttpPath: true});
```

Comment:
I believe I found the right solution now. Please review the updated PR. The mentioned unit test is still there, but it needs to test the normalization through a `RestAdapter`, as the logic to normalize the HTTP paths is there now. I've adjusted the test accordingly. From my tests, all these scenarios of different values for `normalizeHttpPath` work as expected now:

| `config.json`  | `model.json` | Result |
| -------------- | ------------- | ------ |
| `undefined`  | `undefined`  | `ModelName/relationshipName` |
| `undefined`  | `false` | `ModelName/relationshipName` |
| `undefined`  | `true`  | `model-name/relationshipName` |
| `false`  | `undefined`  | `ModelName/relationshipName` |
| `true`  | `undefined`  | `model-name/relationship-name` |
| `false`  | `false`  | `ModelName/relationshipName` |
| `false`  | `true`  | `model-name/relationshipName` |
| `true`  | `false`  | `ModelName/relationship-name` |
| `true`  | `true`  | `model-name/relationship-name` |